contracts/mock/providers/compound/CERC20Mock.sol

```diff
@@ -40,18 +40,37 @@ contract CERC20Mock is ERC20Mock {
         return NO_ERROR;
     }
 
-    function redeem(uint256 redeemTokens) external returns (uint256) {
-        underlying;
-        redeemTokens;
-        _mockChange();
+    function redeem(uint256 redeemAmount) external returns (uint256) {
+        if (SIMULATE_COMPOUND_REDEEM_UNDERLYING_ERROR == redeemAmount) {
+            return NO_ERROR;
+        }
+        if (balanceOf(msg.sender) < redeemAmount) {
+            return 13;
+        }
+
+        uint256 tokenAmount = _getTokensAmount(redeemAmount);
+        require(super.transfer(msg.sender, tokenAmount), "UNDERLYING_TRANSFER_FAILED");
+        super.burn(redeemAmount);
+        require(
+            ((ERC20Mock(address(underlying))).mint(msg.sender, redeemAmount)),
+            "UNDERLYING_MINT_FAILED"
+        );
+        if (SIMULATE_COMPOUND_REDEEM_UNDERLYING_RETURN_ERROR == redeemAmount) {
+            return RETURN_ERROR;
+        }
         return NO_ERROR;
     }
 
     // https://compound.finance/docs/ctokens#redeem-underlying
+    // TODO: this logic needs to taken the amount passed in and transfer that amount of the underlying asset, then convert it to a ctoken amount to burn.
```

Comment:
Will this TODO be fixed on this PR?

Reply:
There are a few things that need to change in this mock file. Not sure if this is the right PR to do it in